interpreter.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+from rich.text import Text
+from expression import Expression
+from extended_list import ExtendedList
+from globals import console
+from tokenizer import Tokenizer
+
+
+class Interpreter:
+
+    @staticmethod
+    def evaluate_token_list(token_list: list[str | Expression]) -> Expression:
+        """
+        evaluates the given list of tokens
+        :parameter token_list: a list of tokens from the tokenizer to evaluate
+        :returns: the value the token list evaluates to
+        """
+
+        i = 0
+        while i < len(token_list):
+            token = token_list[i]
+            if token == "(":
+                closing_brace_index = ExtendedList(token_list).get_closing_brace_index(i)
+                token_list[i] = Interpreter.evaluate_token_list(token_list[i + 1: closing_brace_index])
+                for j in range(closing_brace_index - i):
+                    del token_list[i + 1]
+            i += 1
+        i = 0
+        while i < len(token_list):
+            token = token_list[i]
+            if str(token.__class__) == "<class 'str'>" and token == "nicht":
```

Comment:
Hab ich mal für dich übrig gelassen...
Typen sollte man im besten Fall nicht vergleichen müssen
Und wenn, dann nicht mit Vergleich des Namens sondern `type(token) is str`